lib/core/client.js

```diff
@@ -1153,6 +1153,41 @@ function write (client, request) {
       .on('drain', onDrain)
       .on('error', onFinished)
       .on('close', onFinished)
+  } else if (util.isIterable(body)) {
+    try {
+      if (contentLength === null) {
+        socket.write(`${header}transfer-encoding: chunked\r\n`, 'ascii')
+      } else {
+        socket.write(`${header}content-length: ${contentLength}\r\n\r\n`, 'ascii')
+      }
+      for (const chunk of body) {
```

Comment:
for await

Reply:
move this part out to a `async function writeIterable` and maybe refactor the above into a `function writeStream`

Reply:
when I make the write method async it breaks a bunch of stuff since write is called synchronously in a while loop. any ideas to change that?

Reply:
Yes... what I wrote above 😊